test/argmaxpool-microkernel-tester.h

```diff
@@ -290,3 +295,1185 @@ class ArgMaxPoolMicrokernelTester {
+      GTEST_SKIP();                                                                                                    \
+    }                                                                                                                  \
+    const size_t channel_scaled_tile = channel_tile * get_batch_scale<datatype>();                                     \
+    size_t _input_offset =                                                                                             \
+      (channel_tile == channel_scaled_tile) ? xnnpack::NextPrime(channel_tile + 1) : channel_scaled_tile + 1;          \
+    const std::string kernel_name = #ukernel;                                                                          \
+    for (size_t _pooling_elements = 2; _pooling_elements < primary_tile; _pooling_elements++) {                        \
+      ArgMaxPoolMicrokernelTester tester;                                                                              \
+      tester.pooling_elements(_pooling_elements);                                                                      \
+      tester.pooling_tile(primary_tile, (incremental_tile != 0) ? incremental_tile : 0);                               \
+      tester.channels(channel_scaled_tile);                                                                            \
+      tester.input_offset(_input_offset);                                                                              \
+      if (kernel_name.find("scalar") != std::string::npos) {                                                           \
+        tester.Test(ukernel, ArgMaxPoolMicrokernelTester::Variant::Scalar);                                            \
+      }                                                                                                                \
+      else {                                                                                                           \
+        tester.Test(ukernel);                                                                                          \
+      }                                                                                                                \
+    }                                                                                                                  \
+  }
+
+#define XNN_TEST_ARGMAXPOOL_CHANNELS_DIV_UNIPASS(                                                                      \
+  ukernel, arch_flags, primary_tile, incremental_tile, channel_tile, vector_tile, datatype, params_type, init_params)  \
+  TEST(ukernel, channels_div_unipass_fulltile)                                                                         \
+  {                                                                                                                    \
+    TEST_REQUIRES_ARCH_FLAGS(arch_flags);                                                                              \
```

Comment:
The bodies of these tests need to be refactored into helper functions. These macros are too big, it's hard to follow and fix compiler errors in these macros.